File: Basic Arrays/lexographic.cpp
#include<bits/stdc++.h>
using namespace std;
void dfs(int cur,int ending)
{   
    if (cur > ending) return;
    cout<<cur<<endl;
    for(int i=0;i<10;i++)
    {
        int new_num = 10*cur + i ;
        // cout << new_num<<endl;
        if (new_num>ending) return;
        dfs(new_num,ending);
    }
}
int main()
{
// dfs(1,20);
for(int i=1;i<10;i++)
{
dfs(i,20);
}
}

File: Basic Arrays/lexographic.py
# final = []
# def dfs(cur,n):
#     global final
#     print(cur)
#     final.append(cur)
#     if cur > n:
#         return
#     for i in range(10):
#         if(10*cur+i>n):
#             return;
#         dfs(10*cur+i,n)

# for i in range(1,10):
#     dfs(i,20)

# print(*final)
from typing import List
def lexicalOrder(n: int) -> List[int]:
    global final
    final = []
    
    def dfs(cur,n):
        print(cur)
        # global final
        # final.append(cur)
        if cur > n:
            return
        for i in range(10):
            if(10*cur+i>n):
                return;
            dfs(10*cur+i,n)
        
    for i in range(1,10):
        dfs(i,n)
    
    return final

lexicalOrder(12)